src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -485,6 +457,123 @@ void RegisterNotification(DocumentChange notification)
             }
         }
 
+        private IEnumerable<Document> GetDataToSend(DocumentsOperationContext docsContext, SubscriptionState subscription, long startEtag, SubscriptionPatchDocument patch, Stopwatch sendingCurrentBatchStopwatch)
+        {
+            
+            var db = TcpConnection.DocumentDatabase;
+            if (subscription.Criteria.IsVersioned)
+            {
+                if (db.DocumentsStorage.VersioningStorage == null || db.DocumentsStorage.VersioningStorage.IsVersioned(subscription.Criteria.Collection) == false)
+                    throw new SubscriptionClosedException("Cannot use a version subscription, database does not support versioning"); // todo: improve exception handling
```

Comment:
Error should be something like: "The database doesn't have versioning enabled for collection ABC"